xee/ext.py

```diff
@@ -231,19 +231,31 @@ def __init__(
     # Parse the dataset bounds from the native projection (either from the CRS
     # or the image geometry) and translate it to the representation that will be
     # used for all internal `computePixels()` calls.
-    try:
-      if isinstance(geometry, ee.Geometry):
+    if geometry is None:
+      try:
+        x_min_0, y_min_0, x_max_0, y_max_0 = self.crs.area_of_use.bounds
+      except AttributeError:
+        # `area_of_use` is probable `None`. Parse the geometry from the first
+        # image instead (calculated in self.get_info())
         x_min_0, y_min_0, x_max_0, y_max_0 = _ee_bounds_to_bounds(
             self.get_info['bounds']
         )
-      else:
-        x_min_0, y_min_0, x_max_0, y_max_0 = self.crs.area_of_use.bounds
-    except AttributeError:
-      # `area_of_use` is probable `None`. Parse the geometry from the first
-      # image instead (calculated in self.get_info())
+    elif isinstance(geometry, ee.Geometry):
       x_min_0, y_min_0, x_max_0, y_max_0 = _ee_bounds_to_bounds(
           self.get_info['bounds']
       )
+    elif isinstance(geometry, Union[List, Tuple, np.ndarray, Sequence]):
```

Comment:
Sorry, I meant can sequence replace the rest.

Reply:
One thing to consider is that sequence will let in a simple string e.g. `isinstance("hotdog", Sequence)` evaluates to True. Still want me to drop all types in favor of Sequence alone? There is a part of me that is also weirded out by even using np.ndarray since that could be an incorrect shape that still evaluates to length four.

Reply:
I think I would like to be more stringent with types and changes this to only `Union[List, Tuple]`.

Reply:
Thanks for checking. SGTM.